### DISCUSSION TITLE
Multiple annotation frames

### DISCUSSION THREAD

**Discussion by smutch** [hkchengrex/XMem#67] [Q&A] [answered]
Firstly, thanks for the fantastic model and putting it out there.

I'm a little unclear about multiple annotation frames. In my particular use case the object I wish to track disappears from view several times for extended periods. Upon disappearing, the model loses track of it. I tried adding annotation frames at points where the object returns to full view, however, this didn't appear to work. I also want to track objects which may only appear midway through the video. Is this type segmentation task possible with xmem or would I need to chunk up the video and process each segment with my target objects present separately?

Apologies if any of what I'm asking is unclear. This is my first foray into CV and ML...

**Comment by hkchengrex** [comment, marked as answer]
1. Multiple annotations frame.
This is not implemented (as no dataset does this), but you can add a bit of your own code to make it work. See https://github.com/hkchengrex/XMem/issues/21

2. Objects appear midway.
That looks like a YouTubeVOS format. You can have annotation for object 1 at the first frame, and annotation for object 2 at some other later frame.

**Reply by smutch** [reply]
Fantastic! That sound like it could work well for us. Thanks for your help!
